test/mocks.go

```diff
@@ -0,0 +1,158 @@
+// Copyright 2016 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package test
+
+import (
+	"bytes"
+	"reflect"
+	"testing"
+
+	"github.com/golang/protobuf/proto"
+
+	"istio.io/manager/model"
+)
+
+type MockRegistry struct {
+	store   map[model.ConfigKey]*model.Config
+	mapping model.KindMap
+}
+
+func NewMockRegistry() model.Registry {
+	return &MockRegistry{
+		store:   make(map[model.ConfigKey]*model.Config),
+		mapping: MockMapping,
+	}
+}
+
+func (r *MockRegistry) Get(key model.ConfigKey) (*model.Config, bool) {
+	out, err := r.store[key]
+	return out, err
+}
+
+func (r *MockRegistry) Put(obj model.Config) error {
+	if err := r.mapping.ValidateConfig(obj); err != nil {
+		return err
+	}
+	r.store[obj.ConfigKey] = &obj
+	return nil
+}
+
+func (r *MockRegistry) Delete(key model.ConfigKey) {
+	delete(r.store, key)
+}
+
+func (r *MockRegistry) List(kind string) []*model.Config {
+	var out = make([]*model.Config, 0)
+	for _, v := range r.store {
+		if v.Kind == kind {
+			out = append(out, v)
+		}
+	}
+	return out
+}
+
+const MockKind = "MockConfig"
```

Comment:
can we move the const and var blocks to the top, and then collapse into const() and var() blocks?